src/jsonrpc.rs

```diff
@@ -84,6 +91,32 @@ fn launch_ipc_client(pool: ThreadPool, infura_project_id: String, ipc_path: &str
     println!("Clean exit");
 }
 
+#[cfg(windows)]
```

Comment:
I'm not a huge fan of simply duplicating all the logic, it seems a bit excessive. Maybe try something like detecting the os and creating the `listener_result` before we launch the ipc client, and then you can pass it in as one of the args?

Reply:
Ok I cleaned this up a bit let me know what you think about this solution